PARCEL_2_RFC.md

```diff
@@ -0,0 +1,1285 @@
+not be part of Parcel 2.
+
+---
+
+## Introduction
+
+Parcel is a compiler for all your code, regardless of the language or toolchain.
+
+Parcel takes all of your files and dependencies, transforms them, and merges
+them together into a smaller set of output files that can be used to run your
+code.
+
+Parcel supports many different languages and file types out of the box, from
+web technologies like HTML, CSS, and JavaScript, to lower level languages like
+Rust, and anything that compiles to WebAssembly (WASM), to assets like images,
+fonts, videos, and more.
+
+Parcel makes your code portable, you can build your code for different
+environments, for the web for your server, or for an app. You can even build
+multiple targets at once and have them live update as you make changes.
+
+Parcel is fast and predictable. It compiles all of your files in isolation in
+parallel inside workers, caching all of them as it goes along. Caches are
+stable across machines and are only affected by the files and configs within
+your project (unless you want to pass specific environment variables).
+
+## Getting Started
+
+Before we get started, you'll need to install Node and Yarn (or npm) and create
+a `package.json` for your project if you haven't already.
+
+```sh
+yarn init
+```
+
+Then with Yarn you can install `parcel` into your app:
+
+```sh
+yarn add --dev parcel
+```
+
+From there you just need to point Parcel at some of your entry files. Like if
+you're building a website, an `index.html` file:
+
+```html
+<!doctype html>
+<html lang="en">
+  <head>
+    <meta charset="utf-8"/>
+    <title>My First Parcel App</title>
+  </head>
+  <body>
+    <h1>Hello, World!</h1>
+  </body>
+</html>
+```
+
+Now if you just run:
+
+```sh
+yarn parcel --serve
+```
+
+You should get a URL that looks something like: `http://localhost:1234/`
+
+Next you can start adding dependencies by specifying them in your code (however
+your language specifies other assets). So for HTML we could create a
+`styles.css` file next to our `index.html` file and include it with a `<link>`
+tag.
+
+```css
+h1 {
+  color: hotpink;
+  font-family: cursive;
+}
+```
+
+```html
+<!doctype html>
+<html lang="en">
+  <head>
+    <meta charset="utf-8"/>
+    <title>My First Parcel App</title>
+    <link rel="stylesheet" href="./styles.css"/>
+  </head>
+  <body>
+    <h1>Hello, World!</h1>
+  </body>
+</html>
+```
+
+As we make the change you should see the website update with your changes
+without even refreshing the page.
+
+## Parcel CLI
+
+The Parcel CLI is built into the main `parcel` package. While you can install
+it globally and run it, it is much better to install it locally into your
+project as a dev dependency.
+
+```sh
+yarn add --dev parcel
+```
+
+You should also add some "scripts" to your `package.json` to run it easier.
+
+```json
+{
+  "name": "my-project",
+  "scripts": {
+    "build": "parcel --production",
+    "start": "parcel --serve"
```

Comment:
Consider using a postversion script for production builds.

Reply:
I don't think that is necessary for this guide